tracer/test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs

```diff
@@ -432,18 +457,62 @@ public string GetTargetFramework()
             return $"net{_major}{_minor}{_patch ?? string.Empty}";
         }
 
-        public MockTracerAgent GetMockAgent(bool useStatsD = false)
+        public void EnableWindowsNamedPipes(string tracePipeName = null, string statsPipeName = null)
         {
-            // Strategy pattern for agent transports goes here
-            var agentPort = TcpPortProvider.GetOpenPort();
-            var agent = new MockTracerAgent(agentPort, useStatsd: useStatsD);
+            TransportType = TestTransports.WindowsNamedPipe;
+        }
+
+        public void EnableDefaultTcp()
+        {
+            TransportType = TestTransports.Tcp;
+        }
+
+        public void EnableUnixDomainSockets()
+        {
+#if NETCOREAPP
+            TransportType = TestTransports.Uds;
+#else
+            // Unsupported
+            TransportType = TestTransports.Tcp;
```

Comment:
Since this is test code, can we throw an InvalidOperationException here so we don't get any false hopes that the call actually enabled UDS?

Reply:
Done